src/core/crypto/p256.ts

```diff
@@ -0,0 +1,252 @@
+   */
+  toUint8Array(): Uint8Array {
+    return this.key.toUint8Array();
+  }
+
+  /**
+   * Get the public key as a hex string with the 0x prefix.
+   *
+   * @returns string representation of the public key
+   */
+  toString(): string {
+    return this.key.toString();
+  }
+
+  /**
+   * Verifies a signed data with a public key
+   *
+   * @param args.message message
+   * @param args.signature The signature
+   * @returns true if the signature is valid
+   */
+  verifySignature(args: { message: HexInput; signature: WebAuthnSignature }): boolean {
+    const { message, signature } = args;
+
+    // Check challenge
+    const { challenge } = signature.getCollectedClientData();
+    const challengeStr = base64url.decode(challenge);
+    if (challengeStr !== message.toString()) {
+      return false;
+    }
+
+    // Get verification data.
+    const verificationData = signature.getVerificationData();
+    const p256Signature = signature.paar.signature;
+    const rawSignature = p256Signature.toUint8Array();
+
+    return p256.verify(rawSignature, verificationData, this.toUint8Array());
+  }
+
+  serialize(serializer: Serializer): void {
+    serializer.serializeBytes(this.key.toUint8Array());
+  }
+
+  static deserialize(deserializer: Deserializer): P256PublicKey {
+    const bytes = deserializer.deserializeBytes();
+    return new P256PublicKey(bytes);
+  }
+
+  static load(deserializer: Deserializer): P256PublicKey {
+    const bytes = deserializer.deserializeBytes();
+    return new P256PublicKey(bytes);
+  }
+}
+
+/**
+ * A P256 ecdsa private key - this is only used for test purposes as signing is done via passkeys
+ */
+export class P256PrivateKey extends PrivateKey {
+  /**
+   * Length of P256 ecdsa private key
+   */
+  static readonly LENGTH: number = 32;
+
+  /**
+   * The private key bytes
+   * @private
+   */
+  private readonly key: Hex;
+
+  /**
+   * Create a new PrivateKey instance from a Uint8Array or String.
+   *
+   * @param hexInput A HexInput (string or Uint8Array)
+   */
+  constructor(hexInput: HexInput) {
+    super();
+
+    const privateKeyHex = Hex.fromHexInput(hexInput);
+    if (privateKeyHex.toUint8Array().length !== P256PrivateKey.LENGTH) {
+      throw new Error(`PrivateKey length should be ${P256PrivateKey.LENGTH}`);
+    }
+
+    this.key = privateKeyHex;
+  }
+
+  /**
+   * Get the private key in bytes (Uint8Array).
+   *
+   * @returns
+   */
+  toUint8Array(): Uint8Array {
+    return this.key.toUint8Array();
+  }
+
+  /**
+   * Get the private key as a hex string with the 0x prefix.
+   *
+   * @returns string representation of the private key
+   */
+  toString(): string {
+    return this.key.toString();
+  }
+
+  /**
+   * Sign the given message with the private key.
+   *
+   * @param message in HexInput format
+   * @returns Signature
+   */
+  sign(message: HexInput): P256Signature {
+    const msgHex = Hex.fromHexInput(message);
+    const sha3Message = sha3_256(msgHex.toUint8Array());
```

Comment:
Whatever we're signing should probably be in bytes at this point.  Is this what we do in the other private keys?